File: mvc-architecture/mvc-arch/src/main/java/org/example/mvc/modelAndView/view/JspViewResolver.java
package org.example.mvc.modelAndView.view;

public class JspViewResolver implements ViewResolver {
    @Override
    public View resolveView(String viewName) {
        if (viewName.startsWith(RedirectView.DEFAULT_REDIRECT_PREFIX))
            return new RedirectView(viewName);

        return new JspView(viewName + ".jsp");
    }
}

File: refactor-to-kotlin/src/main/kotlin/org/example/mvc/view/resolver/JspViewResolver.kt
package org.example.mvc.view.resolver

import org.example.mvc.view.JspView
import org.example.mvc.view.RedirectView
import org.example.mvc.view.View

const val DEFAULT_REDIRECT_PREFIX = "redirect:"

class JspViewResolver : ViewResolver {
    override fun resolveView(viewName: String): View {
        if (viewName.startsWith(DEFAULT_REDIRECT_PREFIX)) {
            return RedirectView(viewName)
        }

        return JspView("${viewName}.jsp")
    }
}